generator/internal/http_annotation_parser.cc

```diff
@@ -0,0 +1,281 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "generator/internal/http_annotation_parser.h"
+#include "google/cloud/internal/absl_str_cat_quiet.h"
+#include "google/cloud/internal/make_status.h"
+#include "google/cloud/log.h"
+#include "absl/strings/match.h"
+
+namespace google {
+namespace cloud {
+namespace generator_internal {
+namespace {
+
+template <typename T>
+struct ParseSuccess {
+  T value;
+  std::size_t end;  // where parsing finished
+};
+template <typename T>
+ParseSuccess<T> MakeParseSuccess(T value, std::size_t end) {
+  return ParseSuccess<T>{std::move(value), end};
+}
+
+template <typename T>
+using ParseResult = StatusOr<ParseSuccess<T>>;
+
+Status ParseError(absl::string_view input, std::size_t offset,
```

Comment:
Seeing as all the parsing code is `ParseXxx()`, to distinguish this I would be tempted to rename it to something like `MakeParseError()` (and to match `MakeParseSuccess()`).

Reply:
Fixed